ibis/expr/analysis.py

```diff
@@ -144,33 +144,6 @@ def substitute(fn, node):
         return node
 
 
-def substitute_parents(node):
```

Comment:
Apparently we don't require it anymore, I'd guess that the functionality of this function is superseded by `sub_immediate_parents()`.